### PR TITLE
Disable pycontextvar test on PyPy

### PR DESCRIPTION
**Pull request by da-woods**
Now that they've implemented the PyContextVar C API it segfaults
instead of just failing. Therefore, move it to the "crash bugs"
list so that it doesn't get run.

### CODE REVIEW

**Comment by mattip** [comment]
Is there a way to mark the skips/crashes for a particular `sys.implementation.version`? The latest release is 7.3.7, this will be fixed for pypy 7.3.8.

**Comment by da-woods** [comment]
> Is there a way to mark the skips/crashes for a particular sys.implementation.version? The latest release is 7.3.7, this will be fixed for pypy 7.3.8.

There's the "decorator to disable the individual function" approach. I'll give that a go...

**Comment by mattip** [comment]
Thanks, and do please ping me if you hit more places where PyPy is lacking

**Comment by mattip** [comment]
One tiny change, if you want, might be to reverse the "if not, else" in the test file to make it "if, else" instead.

```
@@ -25,11 +25,11 @@ def test_get_value(var, default=NOTHING):
     >>> test_get_value(CVAR_WITH_DEFAULT)
     'DEFAULT'
     >>> test_get_value(CVAR_WITH_DEFAULT, "default")
     'DEFAULT'
     """
-    return get_value(var, default) if default is not NOTHING else get_value(var)
+    return get_value(var) if default is NOTHING else get_value(var, default)
 
 
 def test_get_value_no_default(var, default=NOTHING):
     """
     >>> test_get_value_no_default(CVAR)
@@ -40,6 +40,6 @@ def test_get_value_no_default(var, default=NOTHING):
     'default'
     >>> test_get_value_no_default(CVAR_WITH_DEFAULT)
     >>> test_get_value_no_default(CVAR_WITH_DEFAULT, "default")
     'default'
     """
-    return get_value_no_default(var, default) if default is not NOTHING else get_value_no_default(var)
+    return get_value_no_default(var) if default is NOTHING else get_value_no_default(var, default)
```

File: tests/run/pycontextvar.pyx [hunk 1]
```diff
@@ -43,3 +52,5 @@ def test_get_value_no_default(var, default=NOTHING):
     'default'
     """
     return get_value_no_default(var, default) if default is not NOTHING else get_value_no_default(var)
+
+__test__ = {}
```
Comment (scoder) [inline tests/run/pycontextvar.pyx:56]: Do we still automatically generate the `__test__` dict? Is this still necessary?

**Comment by scoder** [comment]
I've merged it to finally get the tests green again. Thanks!

File: tests/run/pycontextvar.pyx (hunk 1 above)
Comment (da-woods) [reply tests/run/pycontextvar.pyx:56]: I think we do still generate the `__test__` dict.

According to the doctest docs:
> CPython implementation detail: Prior to version 3.4, extension modules written in C were not fully searched by doctest.

I think that means it's still necessary for Python 2.7

**Comment by da-woods** [comment]
>  to finally get the tests green again. 

Unfortunately another segfault appeared to spoil my moment of triumph. Which raises the question: is there any point in running the "PyPy bugs" tests as part of the CI?

I originally left it in with the idea that it would help see what's improving. But I don't think it's actually very easy to pick out any tests that have started passing so it isn't really serving much purpose.

**Comment by mattip** [comment]
I have set up a [binary-testing](https://github.com/pypy/binary-testing/blob/main/.github/workflows/cython.yml) repo to run cython tests against pypy HEAD once a week. Perhaps that would be a better home to test PyPy progress

**Comment by da-woods** [comment]
> I have set up a [binary-testing](https://github.com/pypy/binary-testing/blob/main/.github/workflows/cython.yml) repo to run cython tests against pypy HEAD once a week. Perhaps that would be a better home to test PyPy progress

Good idea. Just to clarify I was only proposing to stop running the known buggy tests - we definitely want to keep running the tests that currently work

**Comment by mattip** [comment]
Hmm. The [test run](https://github.com/pypy/binary-testing/actions/runs/1507414373) for that repo finishes way too fast (after about a minute). I [am running](https://github.com/pypy/binary-testing/blob/main/.github/workflows/cython.yml#L85), any thoughts what I am missing?
```
python runtests.py -vv --no-code-style -x Debugger --backends=${{ matrix.bbackend }} -j7
```

**Comment by da-woods** [comment]
@mattip  You don't mean `matrix.backend` do you? (one less `b`).

The other comment (which I don't think is your real issue) is that I think we skip `python setup.py build_ext -i` on pypy. I'm not sure if that hugely matters but Cython doesn't _need_ to be compiled, and you might be better off letting PyPy's JIT compiler at it rather than building it.

**Pull request merged**